### PR TITLE
Int: Fix bunch of pyright issues

### PR DESCRIPTION
**Pull request by SebastianMorawiec**
# The problem

We had to add tons of files to filterlist for pyright to pass.

# This PR's solution

I went one by one on some of the files and fixed bunch of individual issues
Removed fixed files from exception list

# Checklist

_Check that this PR satisfies the following items:_

- [x] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [x] The PR's title is prefixed with `<feat/fix/chore/imp[rovement]/int[ernal]/docs>[!]:`
- [x] The PR is linked to a JIRA ticket (if there's no suitable ticket, check the box).

### CODE REVIEW

File: src/orquestra/sdk/_base/_logs/_interfaces.py [hunk 1]
```diff
@@ -12,8 +12,8 @@
 
 @dataclass(frozen=True)
 class LogOutput:
-    out: t.Sequence[str]
-    err: t.Sequence[str]
```
Comment (SebastianMorawiec) [inline src/orquestra/sdk/_base/_logs/_interfaces.py:16]: A comment on that change: Its good to have general interface, but we later in the code used + operator which is specifically defined for the lists.

**Review by alexjuda — APPROVED** [review]
🔥

File: src/orquestra/sdk/_base/cli/_arg_resolvers.py [hunk 2]
```diff
@@ -438,7 +438,7 @@ def resolve_limit(
                     "Enter maximum number of results to display. "
                     "If 'None', all results will be displayed."
                 ),
-                default="None",
+                default=None,
```
Comment (alexjuda) [inline src/orquestra/sdk/_base/cli/_arg_resolvers.py:441]: I just saw it mentioned in the tests. IIRC something's wrong with the overall typing here, as "None" should be an accepted output (as much as I don't like this)

Comment (SebastianMorawiec) [reply src/orquestra/sdk/_base/cli/_arg_resolvers.py:441]: If I wasnt mistaken, the correct type here is Optional[int], so no idea how that would work with string type passed out tbh.
None type seems to be better anyway

**Pull request merged**